ArtOfIllusion/src/artofillusion/ViewerCanvas.java

```diff
@@ -77,6 +80,7 @@ Changes copyright (C) 2016-2018 by Maksim Khramov
     }
   }
 
+  public static int coordinateAxisSize = 30;
```

Comment:
Lets make sure we get this right, as it's part of the API of what we can consider an AOI "foundational" class:

Making this a `static` means that any time you change this value through a plugin or script, it automatically applies to *all* `ViewerCanvas`es. Is this the intent?

Also, do we want to hard-code a default here, or should it be an added preference option?

Reply:
> static ..Is this the intent?

Yes it is.

> hard-code a default here, or should it be an added preference option?

I made it so that it's easy to add to preferences but so far we have been living with a hard coded thing. I'd say that the preferences dialog is too crowded already and needs redesign first.

Short answers above, but I can tell more if you wish. :)

Reply:
Okay, you've got me convinced about `static`. 

Also a good point about about the Preferences Dialog. I suspect that it could afford to grow some tabs!

However, if we do, eventually, move this over to preferences, we would have a dilema: a `public` field cannot be made private without marking the API as a breaking change, so it would be tempting to leave the field as-is. But that leaves us open to situations where a script, etc. changes the value, taking it out of sync with the preferences.

So: Other than for testing purposes, do we really have cause to make this value available to every script or plugin that comes along?

Reply:
So let's make it private.

As a matter of fact I don't think I know anybody who would have ever changed the factory setting of the orientation indicator even if available. Some of the usual things I have seen in preferences would be show/don't, in which corner and size. Everybody just switches it shown (if originally not) and lives happily ever after.

Reply:
>So let's make it private.

Let's do it. However, I think that it actually belongs down in the `CoordinateAxes` class, (Will need to be marked `static final` to compile) as it is only used in that code.

If we do do eventually move it into preferences, the inner class can check the preference just as easily as the outer can. However, you are right that many options don't really get used, which is another way of saying that they may not be necessary.